device/esp_tinyusb/test_apps/teardown_device/pytest_teardown_device.py

```diff
@@ -0,0 +1,92 @@
+# SPDX-FileCopyrightText: 2022-2024 Espressif Systems (Shanghai) CO LTD
+# SPDX-License-Identifier: Apache-2.0
+
+import pytest
+from pytest_embedded_idf.dut import IdfDut
+import serial
+# import serial.tools.list_ports
+from serial.tools.list_ports import comports
+# import time
+from time import sleep
+from time import time
+
+
+@pytest.mark.esp32s2
+@pytest.mark.esp32s3
+@pytest.mark.esp32p4
+@pytest.mark.usb_device
+
+def find_tusb_cdc(vid, pid):
+    ports = comports()
+    for cdc in ports:
+        if cdc.vid == vid and cdc.pid == pid:
+            return cdc.device
+    return None
+
+def wait_for_tusb_cdc(vid, pid, timeout=30):
+    start_time = time()
+    while time() - start_time < timeout:
+        tusb_cdc = find_tusb_cdc(vid, pid)
+        if tusb_cdc:
+            return tusb_cdc
+        sleep(0.5)  # Check every 0.5 seconds
+    return None  
+
+def teardown_device(amount):
+    TUSB_VID = 0x303A  # Espressif TinyUSB VID
+    TUSB_PID = 0x4002  # Espressif TinyUSB VID
+
+    # Command to send and expected response
+    COMMAND = "teardown"
+    EXPECTED_RESPONSE = "teardown"
+
+    # Number of iterations, must be equal to ITERATIONS in the test application
+    ITERATIONS = amount
+
+    for i in range(ITERATIONS):
+        print(f"Iteration {i+1} of {ITERATIONS}")
+
+        # Wait for the device to appear
+        print("Waiting for the device to connect...")
+        tusb_cdc = wait_for_tusb_cdc(TUSB_VID, TUSB_PID)
+        if not tusb_cdc:
+            print("Error: Device did not appear within the timeout period.")
+            break
```

Comment:
Maybe, consider adding assert here as well. Otherwise, if the device does not appear, the test passes, which is incorrect.

```suggestion
        if not tusb_cdc:
            print("Error: Device did not appear within the timeout period.")
            assert True
```